stm32: dma: Simplify handling of Kconfig symbols

erwango opened the pull request:
In order to simplify the handling of DMA_STM32_V1/V2 and DMAMUX_STM32
symbols, set them directly based on related compatible status.

As a preliminary step, instantiate matching compatibles.

erwango committed:
dts/arm/st: Split "st,stm32-dma" compatible into -v1 and -v2

2 versions of DMA hardware blocks could be found across stm32 series.
In order to simplify the handling of matching Kconfig symbols,
make this visible in dts files by creating "st,stm32-dma-v1" and
"st,stm32-dma-v2" and set them accordingly in dtsi files.

Duplicate and update related bindings to reflect that new state.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/dma: stm32: Base Konfig symbols on dts compatible status

In order to simplify the handling of DMA_STM32_V1/V2 and DMAMUX_STM32
symbols, set them directly based on related compatible status.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
^^@shlomow

tagunil commented on dts/bindings/dma/st,stm32-dma-v2.yaml:
```diff
@@ -0,0 +1,106 @@
+         label = "DMA_2";
+        };
+
+  For the client part, example for stm32l47 on DMA1 instance
```
l476?

erwango replied:
Done, txs for the catch

tagunil commented on dts/bindings/dma/st,stm32-dma-v2.yaml:
```diff
@@ -0,0 +1,106 @@
+
+    st,mem2mem:
+      type: boolean
+      description: If the DMA controller V1 supports memory to memory transfer
```
V2?

shlomow commented on drivers/dma/Kconfig.stm32:
```diff
@@ -15,20 +19,20 @@ if DMA_STM32
 	bool
-	depends on SOC_SERIES_STM32F2X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
+	default $(DT_COMPAT_ST_STM32_DMA_V1)
 	help
```
I think the help strings should be changed too

erwango replied:
Indeed, done

shlomow commented on dts/arm/st/h7/stm32h7.dtsi:
```diff
@@ -678,7 +678,7 @@
+			compatible = "st,stm32-dma-v2";
 			#dma-cells = <4>;
 			reg = <0x40020400 0x400>;
 			interrupts = <11 0>, <12 0>, <13 0>, <14 0>, <15 0>, <16 0>,
```
not really related to this, but the `interrupts` property should be `interrupts = <56 0>, <57 0>, <58 0>, <59 0>, <60 0>, <68 0>, <69 0>, <70 0>;`

erwango replied:
Thx for catching that. I've fixed this as well.

FRASTM commented on soc/arm/st_stm32/stm32l4/Kconfig.defconfig.series:
```diff
@@ -16,10 +16,5 @@ config SOC_SERIES
 config STM32_LPTIM_TIMER
 	bool
 	default y
-	depends on PM
```
why removing "depends on PM" there

erwango replied:
Not related indeed. Don't know how this ended up here. Will fix

FRASTM reviewed (approved):
else LGTM

erwango commented on dts/bindings/dma/st,stm32-dma-v2.yaml:
```diff
@@ -0,0 +1,106 @@
+       	       0x2: high
+       	       0x3: very high
+    4. features: A 32bit bitfield value specifying DMA features
+        -bit 0-1: DMA FIFO threshold selection (not USED for DMA V2)
```
@FRASTM, I'm not sure about this part.
Does it mean that "features" has no use for V2 ?
(Even if not used, it should be kept for to simplify things at users level, but then I need to make it clear).

FRASTM replied:
yes fifo threshold is not used in case of DMA V2

erwango replied:
ok, txs. I've made it clear in the binding.

erwango committed:
dts/arm/st: stm32h7: Fix dma2 interrupts on STM32H7 series

Fix DMA2 Interrupts numbers likely due to copy/paste issue.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

shlomow commented on dts/arm/st/h7/stm32h7.dtsi:
```diff
@@ -667,7 +667,7 @@
 
 		dma1: dma@40020000 {
-			compatible = "st,stm32-dma";
+			compatible = "st,stm32-dma-v2";
```
h7 is v1, no?

erwango replied:
Good catch, will fix

erwango replied:
Done

mbolivar-nordic commented on drivers/dma/Kconfig.stm32:
```diff
@@ -4,6 +4,10 @@
 # Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_ST_STM32_DMA_V1 := $(dt_compat_enabled,$(st,stm32-dma-v1))
+DT_COMPAT_ST_STM32_DMA_V2 := $(dt_compat_enabled,$(st,stm32-dma-v2))
+DT_COMPAT_ST_STM32_DMAMUX := $(dt_compat_enabled,$(st,stm32-dmamux))
```
The `$(st,stm32-dma-v1)` part doesn't look right to me. There's no kconfigfunction named `st`.

Are you sure about this?

erwango replied:
Late untested fix. Txs !

erwango committed:
dts/arm: st: Fix use of "st,mem2mem" dma property

"st,mem2mem" property is supposed to be limited to dma-v1.
Remove its use in dma-v2 components.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
